Modelos de Valor del Dinero en el Tiempo
===

Notas de clase sobre ingeniería economica avanzada usando Python

**Juan David Velásquez Henao**  
jdvelasq@unal.edu.co   
Universidad Nacional de Colombia, Sede Medellín  
Facultad de Minas  
Medellín, Colombia 

**Software utilizado**
>Este es un documento interactivo escrito como un notebook de [Jupyter](http://jupyter.org/) , en el cual se presenta un tutorial sobre finanzas corporativas usando Python. Los notebooks de Jupyter permiten incoporar simultáneamente código, texto, gráficos y ecuaciones. El código presentado en este notebook puede ejecutarse en los sistemas operativos Linux y OS X.  

>Haga click [aquí](https://github.com/jdvelasq/guias-de-instalacion) para obtener instrucciones detalladas sobre como instalar Jupyter en Windows y Mac OS X.

>Descargue la última versión de este documento a su disco duro; luego, carguelo y ejecutelo en línea en Try [Jupyter!](https://try.jupyter.org/)



#### Contenido

>* 

**Bibliografía** 

> * [1] SAS/ETS 14.1 User's Guide, 2015.   
* [2] **hp** 12c platinum financial calculator. User's guide. 
* [3] HP Business Consultant II Owner's manual.
* [4] C.S. Park and G.P. Sharp-Bette. Advanced Engineering Economics. John Wiley & Sons, Inc., 1990.

# Diagrama de flujo de dinero

Para la realización de cálculos financieros, el flujo de dinero en el tiempo es representado mediante el diagrama de flujo de dinero. En este, las flechas hacia arriba representan dinero recibido, mientras que las flechas hacia abajo representan dinero pagado.

<img src="images/cash-flow.png" width=600>

# Valor del dinero en el tiempo

[Contenido](#Contenido)

Un peso hoy vale más que un peso mañana, debido a que un peso hoy se puede invertir (hoy) para obtener intereses.

<img src="images/cashflow-1-period.png" width=600>

De forma general:

* $PV$ -- Valor actual
* $FV$ -- valor futuro (FV) 
* $r$ -- tasa de interés  

$$FV = PV \times (1 + r)$$   

$$PV= \left( \displaystyle\frac{1}{1+r}\right) \times FV $$

Componentes de la tasa de interés:

 $$1+r = \left( \displaystyle\ 1 + r_\alpha\right) \left( \displaystyle\ 1 + f\right) \left( \displaystyle\ 1 + r_\pi\right)$$
 

* $r_{α}$ -- Interés real

* $f$ -- Inflación

* $r_{π}$ -- Componente de riesgo

# Interés simple

En el interés simple se reciben directamente los intereses del capital sin reinvertirlos. Es típico en algunos tipos de préstamos.

**Ejemplo.--** [2, pag. 43] Un amigo necesita un préstamo de 450 por 60 días. Usted le presta a un interés simple del 7%, calculado sobre una base anual de 360 días. Cuál es la cantidad total que usted debe recibir al final de periodo?

<img src="images/simple-interest.png" width=350>

In [1]:
## cálculo manual
-450 * (1 + 0.07 * 60 / 360)

-455.25

**Ejemplo.--** [2, pag. 43] Realice el mismo ejemplo anterior sobre una base de 365 días.

In [2]:
## cálculo manual
-450 * (1 + 0.07 * 60 / 365)

-455.17808219178085

**Ejemplo.--** ¿Qué cantidad de dinero se poseerá después de prestar \$ 2300 al 25% de interés simple anual durante 3 años? 

In [3]:
## cálculo manual
2300 * (1 + 0.25 * 3)

4025.0

---

**Ejercicio.--** ¿Qué cantidad de dinero se poseerá después de prestar \$ 10000 al 30% de interés simple anual durante 2 años? (R/ \$ 16000)

---

# Interés compuesto

[Contenido](#Contenido)

En el interés simple se reciben directamente los pagos del interés. En el interés compuesto, los intereses se suman al capital $(P)$, tal que se reciben intereses sobre los intereses de periodos pasados.

![alt text](images/interes-simple-y-compuesto.png "intereses")

Diferencia:
$$P[(1+i)^N-1]-iNP = P[(1+i)^N-(1+iN)]$$


# Concepto de equivalencia financiera

[Contenido](#Contenido)

Dos flujos de fondos son equivalentes a la tasa de interés $r$ si el uno puede ser convertido en el otro usando las tranformaciones apropiadas de interés compuesto.

<img src="images/equivalencia.png" width=700>

# Modelos de valor del dinero en el tiempo

[Contenido](#Contenido)

## Equivalencia entre un pago actual y un pago futuro.

[Contenido](#Contenido)

<img src="images/equiv-pago-actual-futuro-disc.png" width=400>

$$F = - P * (1+r)^n$$

$$P = - F * (1 + r)^{-n} = -\frac{F}{(1+r)^n}$$

Para resolver este tipo de problemas se requieren tres de las cuatro variables de la formula.

---

**Ejercicio.--** Exprese $r$ en función de $P$, $F$ y $n$.

**Ejercicio.--** Exprese $n$ en función de $P$, $F$ y $r$.

---

**Ejemplo.--** ¿Cuánto dinero se deberá invertir hoy si se quiere obtener al final de 5 años \$ 7800 a un interés trimestral del 2%?

<img src="images/sesion-1-ejemplo-1.png" width=400>

In [4]:
-7800 / ((1 + 0.02) ** 20)

-5249.176398242848

**Ejemplo.--** Cual es el valor futuro de \$ 730 dentro de un año con un interés mensual del 2.3%

In [5]:
730 * (1 + 0.023)**12

959.0261838317085

## Equivalencia entre un pago actual y una serie de pagos iguales perpetuos.

[Contenido](#Contenido)

<img src="images/equiv-pmt-inf.png" width=300>

Se calcula $P$ como el valor presente de los pagos iguales $A$:

$$ P=\frac{A}{(1+r)} + \frac{A}{(1+r)^2} + ... $$

Sumando $A$ a ambos lados de la ecuación:

$$P+A~=~A \left [  1 + \left(\frac{1}{1+r}\right) + \left(\frac{1}{1+r}\right) ^2 + ... \right ]~=~A\left [ \frac{1}{1-\left(\frac{1}{1+r}\right)}\right] ~=~ A + \frac{A}{r}$$

Al despejar $P$ se obtiene:

$$P=\frac{A}{r}$$

## Equivalencia pagos iguales finitos.

[Contenido](#Contenido)

<img src="images/equiv-pmt-finitos.png" width=500>

Las formulas financieras para anualidades se obtienen al considerar la resta de dos anualidades infinitas; la primera se inicia en el instánte 0 y la segunda en el instánte $n$:   

$$F~=~ P\times(1+r)^n - P^* ~=~\frac{A}{r} \times (1+r)^n - \frac{A}{r} ~=~ A \left [ \frac{(1 + r)^n -1}{r} \right ]$$

---

**Ejercicio.--** Derive la ecuación para calcular el valor presente ($P$) en función de $A$, $r$ y $n$.

**Ejercicio.--** Escriba $A$ en función de $P$, $r$ y $n$.

**Ejercicio.--** Escriba $r$ en función de $P$, $n$ y $A$.

**Ejercicio.--** Escriba $n$ en función de $n$ en función de $P$, $A$ y $r$.

**Ejercicio.--** Cuál tasa de interés hace que los siguientes flujos sean equivalentes?

* Recibir \$ 1000 hoy.

* Recibir \$ 600 al final del periodo 1 y otros \$ 600 al final del periodo 2.

<img src="images/sesion-1-ejemplo-2.png" width=400>

---

## Modo de pago de anualidades finitas

[Contenido](#Contenido)

En el caso de pagos iguales periodicos, es posible establecer el pago al principio o al final del periodo. 

<img src="images/payment-mode.png" width=500>

---

**Ejercicio.--** Derive las ecuaciones de equivalencia entre una anualidad finita anticipada y su valor futuro equivalente y despeje $A$, $F$, $r$ y $n$.

<img src="images/eqiv-pmt-ant-finita.png" width=400>

---

## Modelo general de equivalencia

Los modelos de valor de dinero en el tiempo (TVM -- time value of money) se basan en los siguientes esquemas.

<img src="images/tvm.png" width=700>

---

**Ejercicio.--** ¿Cuál es el valor de A para que los dos flujos sean equivalentes usando una tasa ($r$) del 8%?

<img src="images/ejercicio-A.png" width=500>

---

# Flujos típicos

[Contenido](#Contenido)

Para resolver un problema, primero se debe identificar el flujo de efectivo típico [2, pag. 48]

<img src="images/flujos-tipicos.png" width=700>

---

**Ejercicio.--** Cuál es el valor presente de un bono (obligación) a dos años, cuyo pago (principal) es de \$ 100, con pagos trimestrales de intereses a una tasa del 10%?

---

# Libreria cashflow.py

**`cashflow`** es una librería para la ejecución de cálculos financieros. Las funciones implementadas son similares a las usadas en Microsoft Excel, las calculadoras financieras y otros softwares similares.

In [6]:
# importa la librería financiera.
# solo es necesario ejecutar la importación una sola vez.
import cashflow as cf

**`tvm`** es un modelo para la realización de cálculos simples del valor del dinero en el tiempo. Este modelo usa variables internas para almacenar la información y funciones para realizar los cálculos.

Nomenclatura para los parámetros:

* `pv` -- valor presente.

* `fv` -- valor futuro.

* `pmt` -- pago periodico.

* `nper` -- cantidad de periodos.

* `rate` -- tasa de interés por periodo.

* `when` -- momento del periodo en que se paga la anualidad: `'end'` (o 0) indica el pago al final del periodo; `'begin'` (o 1) indica el pago al principio del periodo.

Nomenclatura para las funciones de equivalencia financiera:

* `PV()` -- calcula el valor presente.

* `FV()` -- calcula el valor futuro.

* `PMT()` -- calcula el pago periodico.

* `NPER()` -- cantidad de periodos.

* `RATE()` -- tasa de interés por periodo.

* `IPMT()` -- retorna un vector con el interés pagado durante cada periodo en cada pago periodico.

* `PPMT()` -- retorna un vector con el abono al capital realizado durante cada pago periodico.

* `BEGBAL()` -- retorna un vector con el balance al inicio de cada periodo.

* `ENDBAL()` -- retorna un vector con el balance al final de cada periodo.

* `AMRT()` -- imprime la tabla de amortizaciones para los cálculos realizados.

Adicionalmente, contiene las siguientes funciones:

* `tolist()` -- devuelve el flujo de caja equivalente como un objeto lista de python. 

* `toCashflow()` --  devuelve el flujo de caja equivalente como un objeto Cashflow, el cual permite realizar otros cálculos financieros adicionales.

A continuación se presentan varios ejemplos de su uso.

**Ejemplo (cuenta de ahorros).--** [3, pág. 88] Se depositan \$ 2000 en una cuenta de ahorros que paga un interés anual del 7.2% (calculado anualmente). Si no se hacen otros depósitos en la cuenta, cuanto tiempo se requiere para que la cuenta tenga \$ 3000? R/ 5.83

<img src="images/sesion-1-ejemplo-4.png" width=350>

In [7]:
m = cf.tvm(rate =     0.072, # tasa de interes
           pv   = -2000,     # valor presente
           fv   = +3000)     # valor futuro
m.NPER()                     # calcula el número de periodos

5.8318434120486415

In [8]:
# Ya que nper es un valor entre 5 y 6, se requieren 6 años 
# para tener un balance de al menos $ 3000.
# El balance al final de los seis años es (R/ 3035.28):
m.update(nper = 6).FV()

3035.2796333197243

**Ejemplo (cuenta de ahorro).--** ¿Cuánto dinero se deberá invertir hoy si se quiere obtener al final de 5 años \$ 7800 a un interés trimestral del 2%? R/ -5249.18

<img src="images/sesion-1-ejemplo-1.png" width=400>

In [9]:
## crea una instancia del modelo y se almacena en una variable
m = cf.tvm(rate =    0.02,   # tasa de interes
           fv   = 7800,      # valor futuro  
           nper =  5*4)      # número de periodos: 5 años * 4 trimestres por año
m.PV()                       # calcula el valor presente

-5249.1763982428483

In [10]:
m # imprime el contenido del modelo

nper:        20
rate:         0.0200
pv:       -5249.18
fv:        7800.00
pmt:          0.00
when: 'end (0)'

In [11]:
## otra forma forma 
cf.tvm(rate =    0.02,
       fv   = 7800,
       nper =  5*4).PV()

-5249.1763982428483

**Ejemplo (valorización).--** Cuál es el valor futuro de \$ 730 dentro de un año con un interés mensual del 2.3%? R/ \$ 959.03 

<img src="images/sesion-1-ejemplo-3.png" width=350>

In [12]:
cf.tvm(rate =  0.023, 
       nper = 12, 
       pv = -730).FV()

959.02618383170852

 **Ejemplo (leasing).--** [3, pág 02] Se hace un leasing por una maquinaria por 4 años (48 meses) con pagos mensuales de \$ 2400; se debe pagar una cuota adicional de \$ 2400 al principio del leasing para reemplazar la última cuota (que ocurre al principio del mes 48). El contrato incluye una opción de compra al final del periodo de leasing por \$ 15000. Cual valor capitalizado del leasing para una tasa del 1.5% mensual? 

<img src="images/sesion-1-ejemplo-5.png" width=350>

In [13]:
# se pagan 47 cuotas de $ 2400 al principio del mes.
x = cf.tvm(nper =     47,
           pmt  =  -2400,
           when = 'begin',
           rate =      0.015)
pv = x.PV() # valor presente de las 47 cuotas al principio del periodo R/ $ 81735.58
pv

81735.575590862747

In [14]:
pv += 2400 # + cuota adicional al principio del leasing  R/ $ 84135.58
pv

84135.575590862747

In [15]:
pv += cf.tvm(nper =     48,
             fv   = -15000,
             rate =      0.015).PV()  # opcion de compra R/ $ 91476.00
pv

91476.001020605399

**Ejemplo (flujo de caja).--** Cuál es el flujo de efectivo para el ejemplo anterior? 

In [16]:
x = cf.tvm(rate = 0.023, nper = 12, pv = -730)
x.FV()
x.tolist()

[-730, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 959.02618383170852]

**Ejemplo (hipoteca).--** [2, pág. 50] Se hará un préstamo de \$ 35000 con un interes del (10.5% / 12). Si se hacen pagos mensuales de \$ 325 al final de cada mes, cuánto tiempo se requiere para cancelar la deuda?  

In [17]:
## crea una instancia del modelo y se almacena en una variable
m = cf.tvm(rate =     0.105/12, # tasa de interes
           pv   = 35000,      # valor presente  
           pmt =   -325,
           when = 'end')      # 
m.NPER()  

327.43926528588929

Si se hacen 327 pagos de \$ 325, cuánto será el pago No. 328? 

In [18]:
# pago en exceso el valor de la deuda
m.update(nper = 328).FV() 

181.89002161088865

In [19]:
m.pmt() + m.fv()  # valor a cancelar en la cuota No. 3

-143.10997838911135

Se se hacen únicamente 327 pagos, cuál es el valor del pago final para cancelar completamente la deuda?

In [20]:
# valor faltante
m.update(nper = 327).FV() 

-141.86862789501902

In [21]:
m.pmt() + m.fv()

-466.86862789501902

**Ejemplo.--** [2, pág. 53] Se abre una cuenta hoy con un depósito de \$ 775. La tasa de interés es de (6.25%/24). Si se siguen realizando depósitos de \$ 50, cuánto tiempo se requiere para alcanzar un saldo de \$ 4000?   

In [22]:
m = cf.tvm(rate =    0.0625/24, # tasa de interes
           pv   = -775,         # deposito inicial  
           pmt  =  -50,         # depositos periodicos
           fv   = 4000,         # saldo final
           when = 'end')        # depositos al final del periodo 
m.NPER()

57.548385170743181

In [23]:
# saldo al final de 58 periodos
m.update(nper = 58).FV()

4027.26560603081

**Ejemplo.--** [2, pág. 55] Qué tasa de interés debe obtenerse para acumular \$ 10000 en 32 periodos si se hace una inversión de \$ 6000? R/ 1.61%

In [24]:
cf.tvm(pv   = -6000,         # deposito inicial
       nper =    32,         # numero de periodos
       fv   = 10000).RATE()  # saldo final

0.016091394922276913

**Ejemplo.--** [2, pág. 57] Si se va a realizar un leasing a una tasa de interés de (5.9%/12) y se deben realizar 48 pagos de \$ 450 y un pago inicial de \$ 1500 al constituirse el crédito, cuál es el monto del préstamo?

In [25]:
cf.tvm(pmt  =  -450,         # pago mensual
       rate =     0.059/12,  # tasa de interés
       nper =    48).PV()    # numero de periodos

19198.600408593775

In [26]:
_ + 1500

20698.600408593775

**Ejemplo.--** [2, pág. 58] Cuánto se puede pagar por una propiedad que generará un flujo neto anual de \$ 17500 durante 5 años, si al final la propiedad se puede vender en \$ 540.000? (la tasa de interés es del 12%)  

In [27]:
cf.tvm(pmt  =  17500,       # pago periodico anual
       fv   = 540000,       # valor de venta
       rate =      0.12,    # tasa de interés
       nper =      5).PV()  # numero de periodos

-369494.08562908118

**Ejemplo.--** [2, pág. 59] Calcule el pago mensual de una hipoteca por \$ 243400 pagada en 348 meses a una tasa del 5.25%/12.

In [28]:
cf.tvm(pv   = 243400,          # monto
       rate =      0.0525/12,  # tasa de interés
       nper =    348).PMT()    # numero de periodos

-1363.286877407623

**Ejemplo.--** [2, pág. 59] Cuánto es el monto periodico que debe consignarse mensualmente en una cuenta de ahorros si el saldo inicial es de \$ 3200, el saldo final es de \$ 60000, la tasa es de 9.75%/2 y plazo es de 30 meses? 

In [29]:
cf.tvm(pv   = -3200,               # apertura
       fv   = 60000,               # saldo futuro
       nper =    30,
       rate =     0.0975/2).PMT() # tasa de interés

-717.44030300066242

**Ejemplo.--** [2, pág. 61] Si se tiene una hipoteca de \$ 243400 con un pago mensual de \$ 1363.29 en 348 meses a una tasa del 5.25%/12, cuál es el pago que debe realizarse en la cuota 60 para cancelar completamente la deuda?

In [30]:
cf.tvm(pv   = 243400,          # monto
       rate =      0.0525/12,  # tasa de interés
       pmt  =  -1363.29,       # pago mensual
       nper =     60).FV()     # numero de periodos

-222975.98179576019

**Ejemplo.--** [2, pág. 61] Si se consignan \$ 50 al principio de cada mes en una nueva cuenta que paga 6.25%/12, cuál es el saldo al final de 24 meses?   

In [31]:
cf.tvm(pv   =      0,          # monto
       rate =      0.0625/12,  # tasa de interés
       pmt  =    -50.0,        # pago mensual
       when = 'begin',         # pago al principio del periodo
       nper =     24).FV()     # numero de periodos

1281.3359267030767

**Ejemplo.--** [2, 62] Se compra una propiedad por \$ 32000. Si se presenta una depreciación del 2% por año, cuál será el valor de la propiedad al final de 6 años? R/ \$ 28346.96

In [32]:
cf.tvm(pv   = -32000,
       rate =     -0.02,   
       nper =      6).FV()

28346.956187647997

**Ejemplo.--** [3, pág. 81] Se está financiando la compra de un carro nuevo con un leasing a tres años a una tasa de interés del 10.5%/12. El precio del carro es de \$ 7250. Se debe realizar un pago inicial de \$ 1500. Cuánto es el pago mensual si los pagos se hacen al final del mes? 

In [33]:
cf.tvm(pv   = 7250-1500,
       rate =         0.105/12,   
       nper =        36).PMT()

-186.88905015194922

In [34]:
# si se desea reducir el pago periodico en $ 10, cual tasa de interés debería obtenerse.
cf.tvm(pv   = 7250-1500,
       pmt =       -176.89,   
       nper =        36).RATE()

0.0056260172700844041

**Ejemplo.--** [3, pág. 82] El pago máximo mensual por una hipoteca a 30 años es de \$ 630. Si usted puede realizar un pago inicial de \$ 12.000 y la tasa es de 11.5%/12, cuál es el precio máximo de la propiedad?

In [35]:
cf.tvm(pmt  =  -630,   
       rate =     0.115/12,
       nper = 30*12).PV()   + 12000

75617.636073352623

**Ejemplo.--** [3, pág. 83] Se constituye una hipoteca a 25 años por un monto de \$ 75250 y una tasa de interés mensual de 13.8%/12. (a) cuál es el pago mensual; (b) Si se anticipa la venta de la propiedad al final de 4 años, cuánto debe pagarse para cancelar la deuda?   

In [36]:
# (a) R/ $ 894.33
cf.tvm(pv  =  75250,   
       rate =     0.138/12,
       nper = 25*12).PMT() 

-894.33055797115355

In [37]:
# (b) R/ $ -73408.81
cf.tvm(pv   = 75250, 
       pmt  =  -894.33,
       rate =     0.138/12,
       nper =  4*12).FV() 

-73408.811623403832

**Ejemplo (ahorro programado).--** Se abre una cuenta ahorros con un depósito inicial de \$ 2000, y posteriormente se hacen depósitos quincenales de \$ 80 durante 15 años. La cuenta paga un interés del 0.346% quincenal. Cuánto dinero habrá en la cuenta después del último depósito? R/ 63988.44 

In [38]:
cf.tvm(pv   =   -2000,
       pmt  =     -80,
       nper = 15*12*2,
       rate =       0.00346).FV()

63988.433987522505

**Ejemplo (leasing).--** Se hace un leasing por 3 años para un carro nuevo que vale hoy \$ 13500, con la opción de comprar el carro por \$ 7500 al final del leasing. Cuánto es el pago mensual si el interés es del 1.16% mensual? (tenga en cuenta que en el leasing se paga la cuota al principio del periodo) R/ $ 288.49

In [39]:
cf.tvm(pv   =   13500,
       fv   =   -7500,
       nper =      36,
       when = 'begin',
       rate =       0.0116).PMT()

-288.4862833217752

**Ejemplo (préstamo).--** Se va a comprar un equipo con un préstamo por \$ 35000 con pagos mensuales de \$ 325 mensuales. Si la tasa de interés es del 0.88% mensual, cuántos periodos se requieren para realizar el pago de la deuda?

In [40]:
x = cf.tvm(pv   =   35000,
           rate =       0.0088,
           pmt  =    -325)
x.NPER() # R/ 336.77

336.76995141193379

In [41]:
# OPCION 1: se hace nper = 337 para calcular cuanto se paga 
# como excedente en la última cuota. R/ $ 74.51
x.update(nper = 337).FV()

74.513808694551699

In [42]:
325 - x.FV()  # Pago final, restando el excedente.  R/ \$ 250.49

250.4861913054483

In [43]:
# OPCION 2: Se hace un pago adicional con la 
# cuota 336 para cancelar el préstamo
x.update(nper = 336).FV()  # este es el faltante para cancelar la deuda 

-248.30114126228727

In [44]:
-325 + x.FV() # pago final: cuota actual + saldo por cancelar

-573.30114126228727

**Ejemplo (cálculo de la tasa de interés).--** Si hoy se ahorran \$ 6000 y sin hacer más depósitos se tienen \$ 10000 al cabo de 8 años, ¿cuál es la tasa de interés, si los intereses son liquidados trimestralmente? R/ 1.61%

In [45]:
cf.tvm(pv   =   -6000,
       fv   =  +10000,
       nper =     4*8).RATE()

0.016091394922276913

**Ejemplo (inversión).--** Se comprarán unas bodegas para arrendamiento. Si el arriendo es de \$ 17500 anual, y las bodegas se venden al final del año 5 por \$ 540000. Cuál es el monto máximo que puede pagarse por las bodegas si la tasa es del 12% anual?  R/ \$ 369494.09

In [46]:
cf.tvm(fv   = 540000,
       pmt  =  17500,
       rate =      0.12,   
       nper =      5).PV()

-369494.08562908118

**Ejemplo (ahorro programado).--** Si al principio de cada mes se ahorran \$ 100 a una tasa del 2% durante 24 meses, ¿cuánto dinero se tendrá al final del mes 24?.

In [47]:
cf.tvm(rate =    0.02,
       nper =   24,
       pmt  = -100,
       when = 'begin').FV()

3103.0299723236531

**Ejemplo (préstamo).--** Se desea determinar el valor de las cuotas mensuales iguales que se deberán pagar por un préstamo de \$ 16700 financiado a 5 años al 2% mensual

In [48]:
cf.tvm(rate  =      0.02,
       pv    =  16700,
       nper  =   5*12).PMT()

-480.42502929096554

**Ejemplo (préstamo con pago final adicional).--** Se realiza una hipoteca por \$ 243400 a 5 años y con una tasa de interes de 0.12% con pagos mensuales de \$ 4000 y un pago adicional en el último mes. ¿cuál es el valor de dicho pago adicional?

In [49]:
cf.tvm(rate   =       0.0012,
       pv     =  243400,
       nper   =    5*12,
       pmt    =   -4000).FV()

-12863.296612548758

**Ejemplo.--** Cuál será el valor futuro de \$100, \$ 200, \$ 300 y \$ 400 en 5 años a una tasa de interés de 3% anual? 

In [50]:
cf.tvm(pv   =  [100, 200, 300, 400], # una de las variables puede ser un vector
       nper =  5,
       rate =  0.03).FV()

[-115.92740743, -231.85481486, -347.78222229, -463.70962972]

**Ejemplo.--** Cuál será el valor futuro de \$100 en 1, 2, 3 y 4 años a una tasa de interés de 3% anual? 

In [51]:
cf.tvm(pv   =  100, # una de las variables puede ser un vector
       nper =    [1, 2, 3, 4],
       rate =    0.03).FV()

[-103.0, -106.08999999999999, -109.2727, -112.55088100000002]

**Ejemplo.--** Cuál será el valor futuro de \$100 si se invierte a un año al 1%, a 2 años al 1.5% y a 3 años al 1.7%, con una capitalización anual? 

In [52]:
cf.tvm(pv   =  100,                          # si varias variables son listas
       nper =    [1,     2,     3],          # todas ellas deben tener la misma
       rate =    [0.010, 0.015, 0.017]).FV() # longitud

[-101.0, -103.02249999999998, -105.18719129999997]

In [53]:
# verificación del último valor
cf.tvm(pv   =  100, 
       nper =    3,
       rate =    0.017).FV()

-105.18719129999997

**Ejemplo.--** Cuál es la amortización para los siguientes préstamos? (`fv` es el pago final residual (como en el leasing)
```
      plazo    5,        5,       6,       7
      pv     100,      110,     110,     105
      fv     -20,      -10,     -20,       0
      tasa     0.020,    0.017,   0.016,   0.017
      
```

In [54]:
cf.tvm(pv   =   [ 100,     110,     110,     105     ],
       fv   =   [ -20,     -10,     -20,       0     ],         
       nper =   [   5,       5,       6,       7     ], 
       rate =   [   0.020,   0.017,   0.016,   0.017 ]).PMT()

[-17.372671528345773,
 -21.201461428344242,
 -16.1711096183316,
 -16.037190189122793]

**Ejemplo (tablas de amortización - Parte 1).--** Construya la tabla de amortización (balance) para un préstamo de  \$ 1000 a 6 meses con pagos mensuales iguales a una tasa de interés del 1% mensual. 

In [55]:
m = cf.tvm(rate   =    0.01,  # tasa de interés
           pv     = 1000,     # monto
           nper   =    6,     # número de meses 
           when   = 'end')    # pagos al final del periodo

In [56]:
# la función AMRT() imprime la tabla de amortizaciones. 
# tenga en cuenta lo siguiente:
#
#   * el desembolso se recibe al final del periodo 0 (inicio del periodo)
#   * las cuotas se empiezan a pagar al final del periodo 1
#   * de la cuota periodica (PMT) se descuentan los intereses sobre saldo
#     y el resto se abona al capital
#   * el interñes se cálcula como el saldo al principio del periodo 
#     por la tasa de interes efectiva del periodo
#   * al llegar a la última cuota, el saldo final es cero
# 
m.AMRT()

   t      Beginning      Periodic     Interest    Principal        Final
          Principal       Payment      Payment    Repayment    Principal
             Amount        Amount                                 Amount
--------------------------------------------------------------------------
   0        1000.00          0.00         0.00         0.00      1000.00
   1        1000.00       -172.55       -10.00      -162.55       837.45
   2         837.45       -172.55        -8.37      -164.17       673.28
   3         673.28       -172.55        -6.73      -165.82       507.46
   4         507.46       -172.55        -5.07      -167.47       339.99
   5         339.99       -172.55        -3.40      -169.15       170.84
   6         170.84       -172.55        -1.71      -170.84         0.00


In [57]:
# cuota mensual pagada al final del mes
m.PMT()   

-172.54836671088106

In [58]:
# Componente de la cuota abonada a capital
m.PPMT()

[0,
 -162.54836671088106,
 -164.17385037798988,
 -165.81558888176977,
 -167.47374477058747,
 -169.14848221829334,
 -170.83996704047627]

In [59]:
# Balance al principio del periodo
m.BEGBAL()

[1000,
 1000,
 837.45163328911894,
 673.27778291112907,
 507.4621940293593,
 339.98844925877182,
 170.83996704047848]

In [60]:
sum(m.PPMT()) # la suma aritmética de los abonos a capital

-999.99999999999773

In [61]:
m.IPMT()  # devuelve la parte de las cuotas correspondiente a intereses

[0,
 -10.0,
 -8.37451633289119,
 -6.7327778291112912,
 -5.0746219402935928,
 -3.3998844925877183,
 -1.7083996704047848]

In [62]:
sum(m.IPMT())  # la suma aritmética de los intereses pagados

-35.290200265288583

**Ejemplo (tablas de amortización - Parte 2).--** CConstruya la tabla de amortizaciones para el caso anterior, pero considerando que las cuotas se pagan al principio del periodo.

In [63]:
cf.tvm(rate   =    0.01,         # tasa de interés
       pv     = 1000,            # monto
       nper   =    6,            # número de meses 
       when   = 'begin').AMRT()  # pagos al principio del periodo

# 
# Este ejemplo permite clarificar como se realizan los cálculos 
# usando las funciones IPMT() y PPMT()
#
#   * se pagan 6 cuotas como en el caso anaterior
#   * no se reciben 1000 si no 829.16 
#   * el interes se paga de forma vencida. note que la primera cuota
#     no tiene componente de intereses
#

   t      Beginning      Periodic     Interest    Principal        Final
          Principal       Payment      Payment    Repayment    Principal
             Amount        Amount                                 Amount
--------------------------------------------------------------------------
   0        1000.00       -170.84         0.00      -170.84       829.16
   1         829.16       -170.84        -8.29      -162.55       666.61
   2         666.61       -170.84        -6.67      -164.17       502.44
   3         502.44       -170.84        -5.02      -165.82       336.62
   4         336.62       -170.84        -3.37      -167.47       169.15
   5         169.15       -170.84        -1.69      -169.15         0.00
   6           0.00          0.00        -0.00         0.00         0.00


In [64]:
## el cálculo es equivalente a prestar $ 829.16 a 5 periodos
cf.tvm(rate   =    0.01,        # tasa de interés
       pv     = 829.16,        # monto
       nper   =    5,          # número de meses 
       when   = 'end').AMRT()  # pagos al final del periodo

   t      Beginning      Periodic     Interest    Principal        Final
          Principal       Payment      Payment    Repayment    Principal
             Amount        Amount                                 Amount
--------------------------------------------------------------------------
   0         829.16          0.00         0.00         0.00       829.16
   1         829.16       -170.84        -8.29      -162.55       666.61
   2         666.61       -170.84        -6.67      -164.17       502.44
   3         502.44       -170.84        -5.02      -165.82       336.62
   4         336.62       -170.84        -3.37      -167.47       169.15
   5         169.15       -170.84        -1.69      -169.15         0.00


**Ejemplo (tablas de amortización - Parte 3).--** La función requiere que se especifique `pv` o `fv` (o ambas) para el cálculo de la cuota (`pmt`). Note que en el siguiente ejemplo todas las columnas son cero.  

In [65]:
cf.tvm(rate    =    0.01,         # tasa de interés
       pmt    = -100,            # monto
       nper   =    6,            # número de meses 
       when   = 'begin').AMRT()  # pagos al final del periodo

   t      Beginning      Periodic     Interest    Principal        Final
          Principal       Payment      Payment    Repayment    Principal
             Amount        Amount                                 Amount
--------------------------------------------------------------------------
   0           0.00         -0.00         0.00        -0.00         0.00
   1           0.00         -0.00        -0.00         0.00         0.00
   2           0.00         -0.00        -0.00         0.00         0.00
   3           0.00         -0.00        -0.00         0.00         0.00
   4           0.00         -0.00        -0.00         0.00         0.00
   5           0.00         -0.00        -0.00         0.00         0.00
   6           0.00          0.00        -0.00         0.00         0.00


**Ejemplo (tablas de amortización - Parte 4).--** Si se hacen consignaciones iguales desde el mes 1 hasta el mes 6, cuanto debe ser la consignacion para tener un total de \$ 600 al final del último mes?

In [66]:
cf.tvm(rate   =    0.01,         # tasa de interés
       fv     =  600,            # monto
       nper   =    6,            # número de meses 
       when   = 'end').AMRT()    # pagos al final del periodo

   t      Beginning      Periodic     Interest    Principal        Final
          Principal       Payment      Payment    Repayment    Principal
             Amount        Amount                                 Amount
--------------------------------------------------------------------------
   0           0.00          0.00         0.00         0.00         0.00
   1           0.00        -97.53         0.00       -97.53       -97.53
   2         -97.53        -97.53         0.98       -98.50      -196.03
   3        -196.03        -97.53         1.96       -99.49      -295.52
   4        -295.52        -97.53         2.96      -100.48      -396.01
   5        -396.01        -97.53         3.96      -101.49      -497.50
   6        -497.50        -97.53         4.97      -102.50      -600.00


In [67]:
cf.tvm(rate   =     0.01,        # tasa de interés
       fv     =  600,            # monto
       nper   =    6,            # número de meses 
       when   = 'begin').AMRT()  # pagos al final del periodo

   t      Beginning      Periodic     Interest    Principal        Final
          Principal       Payment      Payment    Repayment    Principal
             Amount        Amount                                 Amount
--------------------------------------------------------------------------
   0           0.00        -96.56         0.00       -96.56       -96.56
   1         -96.56        -96.56         0.97       -97.53      -194.09
   2        -194.09        -96.56         1.94       -98.50      -292.60
   3        -292.60        -96.56         2.93       -99.49      -392.09
   4        -392.09        -96.56         3.92      -100.48      -492.57
   5        -492.57        -96.56         4.93      -101.49      -594.06
   6        -594.06          0.00         5.94        -5.94      -600.00


In [68]:
cf.tvm(rate   =     0.01,        # tasa de interés
       pv     = 1000,            # deuda existente
       fv     =  600,            # monto
       nper   =    6,            # número de meses 
       when   = 'begin').AMRT()  # pagos al final del periodo

   t      Beginning      Periodic     Interest    Principal        Final
          Principal       Payment      Payment    Repayment    Principal
             Amount        Amount                                 Amount
--------------------------------------------------------------------------
   0        1000.00       -267.40         0.00      -267.40       732.60
   1         732.60       -267.40        -7.33      -260.08       472.52
   2         472.52       -267.40        -4.73      -262.68       209.84
   3         209.84       -267.40        -2.10      -265.30       -55.46
   4         -55.46       -267.40         0.55      -267.96      -323.42
   5        -323.42       -267.40         3.23      -270.64      -594.06
   6        -594.06          0.00         5.94        -5.94      -600.00


In [69]:
cf.tvm(rate    =    0.01,        # tasa de interés
       pv      = -100,
       fv     =   600,           # monto
       nper   =     6,           # número de meses 
       when   = 'begin').PPMT()

[-79.479389460832223,
 -81.274183355440542,
 -82.086925188994954,
 -82.907794440884899,
 -83.736872385293751,
 -84.574241109146683,
 -5.9405940594059308]

---

# Ejercicios

**Ejercicio.--** Si un banco promete una tasa anual del 8% ¿Cuál será el valor que se obtendrá al final de 3 años de una inversion de \$ 4500? (R/ \$ 5668.70)

**Ejercicio.--** ¿Cuál es el valor presente de un pago único de \$ 1500 recibido dentro de 2 años si la tasa de interés es de 2.7% mensual? (R/ \$ -791.41)

**Ejercicio.--** ¿Cuál es el valor futuro de \$ 609 dentro de 2 años a una tasa de interes mensual de 2%? (R/ $ 979.54) 

**Ejercicio.--** ¿Cuál es el valor presente de un pago único de \$ 890 recibido dentro de 6 años a una tasa de interés  trimestral de 2.7%? (R/ \$ 469.57)   

**Ejercicio.--** ¿Qué cantidad de dinero se poseerá después de prestar \$ 2300 al 25% de interés compuesto anual durante 3 años?

**Ejercicio.--** ¿Qué cantidad de dinero se poseerá después de prestar \$ 10000 al 30% de interés anual durante 2 años? (R/ \$ 16900)

**Ejercicio.--** ¿cuánto dinero se deberá abonar mensualmente (al principio del mes) si se desea reunir \$ 28700 al final de 5 años y los ahorros rentan el 4% mensuales? (R/ \$ 120.59)

**Ejercicio.--**  Se decide ahorrar mensualmente \$ 900 los cuales depositará al principio de cada mes en una entidad
financiera que paga un interés del 2.5% mensual. ¿Cuánto habrá acumulado al cabo de 2 años? (R/ \$ 29114.13)  

**Ejercicio.--** Suponga que usted quiere asegurarse de que puede financiar la educación universitaria de su hija dentro de 14 años a partir de hoy. Su hija retirará \$ 500 al inicio de cada mes de una cuenta de ahorros durante 4 años. ¿Cuánto se tiene que depositar hoy en la cuenta si el interés es del 1% mensual?

**Ejercicio.--** Calcule la cantidad de pago en 29 años de una hipoteca de \$ 243400 con un interés anual de 5.25%.

**Ejercicio.--** Mirando hacia la jubilación, usted desea acumular \$ 60000 después de 15 años haciendo depósitos en una cuenta que paga 5.3% de interés semestral. Usted abre la cuenta con un depósito de \$ 3200 con la intención de hacer depósitos semestrales, comenzando seis meses más tarde a partir de sus ingresos. Calcular la cantidad de dinero que se deberá depositar semestralmente (no se consigna al final del último semestre). 

**Ejercicio.--** Se va a realizar un préstamo de \$ 35000 a un interés de 0.91% (mensual) y con pagos mensuales de \$ 325 al final de cada mes. ¿Cuántos meses se requieren para pagar el préstamo?.  

---

[Contenido](#Contenido)